apollo-http-service/src/main/java/com/spotify/apollo/httpservice/HttpServiceModule.java

```diff
@@ -44,14 +53,23 @@ public static HttpServiceModule create(AppInit appInit) {
   @Override
   protected void configure() {
     bindAppInit();
+    bindClientDecoratorComparator();
 
     install(ApolloEnvironmentModule.create());
+    install(new MetaModule("apollo-http"));
   }
 
   private void bindAppInit() {
     bind(Initializer.class).toInstance(env -> env.initialize(appInit));
   }
 
+  private void bindClientDecoratorComparator() {
+    // ensure that clients track outgoing calls first in the chain
+    bind(new TypeLiteral<Comparator<ClientDecorator.Id>>() { })
+        .toInstance(new ListBasedComparator(singletonList(MetaModule.OUTGOING_CALLS))
```

Comment:
There may and often will be more than one. What this means is that 'apply the outgoing calls decorator first, then any others in random order'.